rero/rero-ils
# patron: the user data comes from the profile

The user profiles was duplicates on the the patron and the profile data.
The synchronization was done in the both size, ether when the user
change his profile and when a professional change the user informations
in the patron editor.

Here the data are modified only on the profile database and synchronized
ony in one direction: from profile to patron.

* New utilities function are proposed to support the current fixtures
users data format.
* All the patron without email fixtures are removed.
* Loans anonymization is removed from the patron update/creation.

__Notes:__

- this PR contains 2 commits to make the review easier
- a new REST API will be create in a next PR to allow a librarian to change the user profile data
- the loan anonymization will be restore in a new PR
- the tests do not need anymore two version for patron fixture with and without email
- the coverage will be increased in the next PRs
- we need to decide will still in the patron JSON or not (if yes: it need a lot of modifications)
 
Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>


## Why are you opening this PR?

- To implement: https://tree.taiga.io/project/rero21-reroils/us/1889?milestone=282105 and https://tree.taiga.io/project/rero21-reroils/us/1905?milestone=282105

## How to test?

- Open an existing patron, change the rero_id and see the user profile data changed in the patron detailed view

## Code review check list
